src/main.ml

```diff
@@ -129,18 +131,22 @@ let main port =
     let sockaddr = Lwt_unix.ADDR_INET(Unix.inet_addr_any, port) in
     Lwt_unix.bind sock sockaddr;
     Lwt_unix.listen sock 5;
-    while_lwt true do
-      Lwt_unix.accept sock
-      >>= fun (fd, _) ->
-      (* Background thread per connection *)
-      let _ =
-        Lwt.catch
-          (fun () -> handle_connection fd)
-          (fun e -> Printf.fprintf stderr "Caught %s\n%!" (Printexc.to_string e); return ())
-        >>= fun () ->
-        Lwt_unix.close fd in
-      return ()
-    done in
+    let rec loop () =
+      begin
+        Lwt_unix.accept sock
+        >>= fun (fd, _) ->
+        (* Background thread per connection *)
+        let _ =
```

Comment:
I don't understand why this would work, or why this worked before. It looks like we're just throwing away the lwt thread that we've constructed, and don't run it at all.